src/IteratorIr.hs

```diff
@@ -15,7 +16,7 @@ module IteratorIr (
 import Control.Lens.Internal.Context
 
 
-data Iterator a b t = It (b -> t) a
+data Iterator i l t = It (l -> t) i
```

Comment:
I would prefer keeping this as is as it is used in many places and the names don’t really matter.